cmd/root/options.go

```diff
@@ -37,26 +40,32 @@ func loadConfigFile(scheme *runtime.Scheme, outConfig *configv1alpha1.ProjectCon
 
 // LoadConfig initializes the controller configuration, optionally overriding the defaults
 // from a provided configuration file.
-func LoadConfig(scheme *runtime.Scheme, configFile string) (*configv1alpha1.ProjectConfig, ctrl.Options, error) {
+func LoadConfig(scheme *runtime.Scheme, configFile string) (*configv1alpha1.ProjectConfig, ctrl.Options, manifestutils.TokenCCOAuthConfig, error) {
 	options := ctrl.Options{Scheme: scheme}
 	ctrlConfig := configv1alpha1.DefaultProjectConfig()
 
 	if configFile == "" {
-		return &ctrlConfig, options, nil
+		return &ctrlConfig, options, manifestutils.TokenCCOAuthConfig{}, nil
 	}
 
 	err := loadConfigFile(scheme, &ctrlConfig, configFile)
 	if err != nil {
-		return nil, options, fmt.Errorf("failed to parse controller manager config file: %w", err)
+		return nil, options, manifestutils.TokenCCOAuthConfig{}, fmt.Errorf("failed to parse controller manager config file: %w", err)
 	}
 
 	err = ctrlConfig.Validate()
 	if err != nil {
-		return nil, options, fmt.Errorf("controller config validation failed: %w", err)
+		return nil, options, manifestutils.TokenCCOAuthConfig{}, fmt.Errorf("controller config validation failed: %w", err)
 	}
 
+	tokenCCOAuth, hasCOOAuthEnv := cloudcredentials.DiscoverTokenCCOAuthConfig()
+
+	// If CCO environment was discovered, override the openshift gateway flag.
+	ctrlConfig.Gates.OpenShift.TokenCCOAuthEnv = hasCOOAuthEnv
```

Comment:
It looks like we don't utilize the feature gate, because it always gets overridden - can we remove the feature gate?
This looks like a good candidate for an `autodiscover` package maybe?